experimenter/experimenter/nimbus-ui/src/components/Summary/index.test.tsx

```diff
@@ -74,6 +74,20 @@ describe("Summary", () => {
       screen.queryByTestId("summary-page-signoff-not-launched"),
     ).toBeInTheDocument();
   });
+  it("updates signoff checklist state correctly", async () => {
+    render(<Subject />);
+    const legalSignoffCheckbox = screen.getByTestId("is-legalsignoff-checkbox");
+    const qaSignoffCheckbox = screen.getByTestId("is-qasignoff-checkbox");
+    const vpSignoffCheckbox = screen.getByTestId("is-vpsignoff-checkbox");
+
+    fireEvent.click(legalSignoffCheckbox);
+    fireEvent.click(qaSignoffCheckbox);
+    fireEvent.click(vpSignoffCheckbox);
```

Comment:
> Most projects have a few use cases for fireEvent, but the majority of the time you should probably use [@testing-library/user-event](https://testing-library.com/docs/user-event/intro).

https://testing-library.com/docs/dom-testing-library/api-events/

We have user-event. Please use that instead. It is likely what was causing your async issues.